staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/sample_and_watermark.go

```diff
@@ -93,8 +93,12 @@ func NewSampleAndWaterMarkHistogramsGenerator(clock clock.PassiveClock, samplePe
 		}}
 }
 
-func (swg *sampleAndWaterMarkObserverGenerator) quantize(when time.Time) int64 {
-	return int64(when.Sub(swg.t0) / swg.samplePeriod)
+func (swg *sampleAndWaterMarkObserverGenerator) quantize(when *time.Time) int64 {
+	diff := when.Sub(swg.t0)
+	if diff < -370*24*time.Hour || diff > 370*24*time.Hour {
```

Comment:
just so I understand, this will log continuously if a server is running for more than a year?

should we precompute these offsets?

Reply:
oh dang, that's right and that is not what I intended.  Fix coming...

Reply:
fixed by force-push to c98c09eb72c